pennylane/ops/qubit/matrix_ops.py

```diff
@@ -183,24 +198,45 @@ def __init__(
                 "The control wires must be different from the wires specified to apply the unitary on."
             )
 
-        U = params[0]
-        target_dim = 2 ** len(wires)
-        if len(U) != target_dim:
-            raise ValueError(f"Input unitary must be of shape {(target_dim, target_dim)}")
-
-        # Saving for the circuit drawer
-        self._target_wires = wires
-        self._control_wires = control_wires
-        self.U = U
+        self._hyperparameters = {
+            "u_wires": wires,
+            "control_wires": control_wires,
+            "control_values": control_values,
+        }
 
-        wires = control_wires + wires
+        total_wires = control_wires + wires
+        super().__init__(*params, wires=total_wires, do_queue=do_queue)
 
-        # If control values unspecified, we control on the all-ones string
-        if not control_values:
-            control_values = "1" * len(control_wires)
+    @property
+    def num_params(self):
+        return 1
 
-        control_int = self._parse_control_values(control_wires, control_values)
-        self.control_values = control_values
+    @staticmethod
+    def compute_matrix(U, control_wires, u_wires, control_values=None):
+        """Canonical matrix representation of the ControlledQubitUnitary operator.
+
+        Args:
+            U (tensor_like): unitary matrix
+            control_wires (Iterable): the control wire(s)
+            u_wires (Iterable): the wire(s) the unitary acts on
+            control_values (str or None): a string of bits representing the state of the control
+                qubits to control on (default is the all 1s state)
+
+        Returns:
+            tensor_like: canonical matrix
+
+        **Example**
+
+        >>> U = np.array([[ 0.94877869,  0.31594146], [-0.31594146,  0.94877869]])
+        >>> qml.ControlledQubitUnitary.compute_matrix(U, control_wires=[1], u_wires=[0], control_values="1")
+        [[ 1.        +0.j  0.        +0.j  0.        +0.j  0.        +0.j]
+         [ 0.        +0.j  1.        +0.j  0.        +0.j  0.        +0.j]
+         [ 0.        +0.j  0.        +0.j  0.94877869+0.j  0.31594146+0.j]
+         [ 0.        +0.j  0.        +0.j -0.31594146+0.j  0.94877869+0.j]]
+        """
```

Comment:
This looks horrendous, but I basically just moved the entire logic from ``init``

Reply:
:+1: